lib/Sema/SemaBounds.cpp

```diff
@@ -2762,8 +2765,10 @@ void Sema::CheckFunctionBodyBoundsDecls(FunctionDecl *FD, Stmt *Body) {
 }
 
 void Sema::CheckTopLevelBoundsDecls(VarDecl *D) {
-  if (!D->isLocalVarDeclOrParm())
-    CheckBoundsDeclarations(*this, nullptr).TraverseVarDecl(D, getCurScope()->isCheckedScope());
+  if (!D->isLocalVarDeclOrParm()) {
+    auto Checker = CheckBoundsDeclarations(*this, nullptr);
```

Comment:
Do we need to update the existed code right after this change? Just like this snippet: 
void Sema::CheckTopLevelBoundsDecls(VarDecl *D) {
  if (!D->isLocalVarDeclOrParm())
    CheckBoundsDeclarations(*this, nullptr).TraverseVarDecl(D, getCurScope()->isCheckedScope());
}